cluster-autoscaler/processors/processors.go

```diff
@@ -73,7 +74,8 @@ type AutoscalingProcessors struct {
 	// * scale-downs per nodegroup
 	// * scale-up failures per nodegroup
 	// * scale-down failures per nodegroup
-	ScaleStateNotifier *nodegroupchange.NodeGroupChangeObserversList
+	ScaleStateNotifier           *nodegroupchange.NodeGroupChangeObserversList
+	ProvisioningRequestProcessor provreq.ProvisoningRequestProcessor
```

Comment:
QQ: how will we add the processor for the atomic for provreq?

Reply:
When we have >1 ProvisioningClass we will need to update processor to contain a map of processors and pick the appropriate processor for each ProvReq or have a list of processors and iterate over each processor and each ProvReq.

Reply:
I would prefer to have a list of processors from the start to avoid changing the logic of the autoscaler multiple times.

Reply:
done